docs/usage_guide/optimization_guide/index.rst

```diff
@@ -0,0 +1,207 @@
+  result = md_df.apply(lambda row: (row[1] + row[2]) / row[3], axis=1).to_numpy().tolist()
+  print(f"Filling a list by using '.apply()' and converting the result to a list: {timer() - t1:.2f}s.")
+  # Possible output: 0.22s.
+
+Use Modin's algebra API to implement custom parallel functions
+""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""
+
+Modin provides a set of low-level parallel-implemented operators which can be used to build most of the
+aggregation functions. These operators are present in the :doc:`algebra module </flow/modin/core/dataframe/algebra>`.
+Modin DataFrame allows for users to use their own aggregations built with this module. Visit the 
+:doc:`appropriate section </flow/modin/core/dataframe/algebra>` of the documentation for the steps to do it.
+
+Do not mix pandas and Modin DataFrames
+""""""""""""""""""""""""""""""""""""""
+
+Although Modin is considered to be a drop-in replacement for pandas, they are not intended to be used together
+in a single flow. Passing a pandas DataFrame as an argument for a Modin's frame method may either slowdown
+the function (because it has to process non-distributed object) or raise an error. You would also get an undefined
+behaviour if pass a Modin frame to pandas methods, pandas identifies Modin's objects as a simple iterable,
+and so can't use its potential.
+
+Copy-pastable example, showing how mixing pandas and Modin frames in a single flow may bottleneck performance:
+
+.. code-block:: python
+
+  import modin.pandas as pd
+  import numpy as np
+  import timeit
+  import pandas
+
+  data = np.random.randint(0, 100, (2 ** 20, 2 ** 2))
+
+  md_df, md_df_copy = pd.DataFrame(data), pd.DataFrame(data)
+  pd_df, pd_df_copy = pandas.DataFrame(data), pandas.DataFrame(data)
+
+  print("concat modin frame + pandas frame:")
+  # Concatenating modin frame + pandas frame using modin '.concat()'
+  # This case is bad because Modin have to process non-distributed pandas object
+  time = timeit.timeit(lambda: pd.concat([md_df, pd_df]), number=10)
+  print(f"\t{time}s.\n")
+  # Possible output: 0.44s.
+
+  print("concat modin frame + modin frame:")
+  # Concatenating modin frame + modin frame using modin '.concat()'
+  # This is an ideal case, Modin is being used as intended
+  time = timeit.timeit(lambda: pd.concat([md_df, md_df_copy]), number=10)
+  print(f"\t{time}s.\n")
+  # Possible output: 0.05s.
+
+  print("concat pandas frame + pandas frame:")
+  # Concatenating pandas frame + pandas frame using pandas '.concat()'
+  time = timeit.timeit(lambda: pandas.concat([pd_df, pd_df_copy]), number=10)
+  print(f"\t{time}s.\n")
+  # Possible output: 0.31s.
+
+  print("concat pandas frame + modin frame:")
+  # Concatenating pandas frame + modin frame using pandas '.concat()'
+  time = timeit.timeit(lambda: pandas.concat([pd_df, md_df]), number=10)
+  print(f"\t{time}s.\n")
+  # Possible output: TypeError
+
+
+Operation-specific settings
```

Comment:
Wondering if the optimization guide should be split into general optimization guidelines and a separate page for operator-specific optimizations.

Reply:
I don't see a problem with putting them on a single page since a user can easily jump through the topics with the right bar. The navigation page with only two links (general opt guide + operation specific) would look empty IMO. We can split the guide into different pages if we end up having much more than only 2 sections.

Reply:
Ok that makes sense. We can keep as a single page.